Add implementation details to tokenized auth (#1443)
openshift/enhancements: enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

- Changing CSV to add CredentialRequest RBAC
- If loading the CredentialRequests from a yaml, place it into a discoverable location in the codebase
- Add a bundle annotation claiming token-based authentication support:
    ```yaml
    apiVersion: operators.coreos.com/v1alpha1
    kind: ClusterServiceVersion
    metadata:
        annotations:
            features.operators.openshift.io/token-auth-aws: "true"
            ...
    ```
- Add a script in the operator description, per supported cloud provider, to help set up the correct role
- (Optional) Add the projected ServiceAccount volume to the Deployment embedded in the CSV;
- When OLM starts the operator with `ROLEARN` env. variable, handle the Secret themselves (like today’s CCO Mint mode,
  read and use the credentials in the Secret whose name they know from the CredentialsRequest and is in the same

>   Namespace.)

Reviewer: aren't we missing the step where the operator is expected to create the CredReq object, also, when the cluster is in STS mode?

Author: I assume that when `ROLEARN` is set, then OLM / console has checked CCO is in STS mode already.

I rephrased the operator behavior, now it should be clearer.

- Changing CSV to add CredentialRequest RBAC
- If loading the CredentialRequests from a yaml, place it into a discoverable location in the codebase
- Add a bundle annotation claiming token-based authentication support:
    ```yaml
    apiVersion: operators.coreos.com/v1alpha1
    kind: ClusterServiceVersion
    metadata:
        annotations:
            features.operators.openshift.io/token-auth-aws: "true"
            ...
    ```
- Add a script in the operator description, per supported cloud provider, to help set up the correct role
- (Optional) Add the projected ServiceAccount volume to the Deployment embedded in the CSV;
- When OLM starts the operator with `ROLEARN` env. variable, handle the cloud credentials *almost* as if CCO is in Mint
  mode:

>   - Create CredentialsRequest, now with `spec.cloudTokenPath` and `spec.providerSpec.stsIAMRoleARN` filed.

Reviewer 2: ```suggestion
  - Create CredentialsRequest, now with `spec.cloudTokenPath` and `spec.providerSpec.stsIAMRoleARN` filled.
```

>   - Wait for Secrets referenced in the CredentialsRequest to be created by CCO.

Reviewer 2: ```suggestion
  - Wait for the Secret referenced in the CredentialsRequest to be created by CCO.
```

>     - Expect the Secrets take some time to create, do not crash and report error when it takes too long.

Reviewer 2: ```suggestion
    - Expect the Secret to take some time to create, do not crash and report an error when it takes too long.
```
